### PR TITLE
Fix several errors reported by clippy.

### PR DESCRIPTION
**Pull request by twittner**
While clippy still reports warnings, this PR fixes things clippy reports as errors.

### CODE REVIEW

File: stores/peerstore/src/peerstore.rs [hunk 1]
```diff
@@ -99,7 +99,6 @@ pub trait PeerAccess {
     /// Sets the TTL of an address of a peer. Adds the address if it is currently unknown.
     ///
     /// Contrary to `add_addr`, this operation is never a no-op.
-    #[inline]
```
Comment (dvdplm) [inline stores/peerstore/src/peerstore.rs]: Was this a clippy error?

File: protocols/ping/src/dial_handler.rs [hunk 2]
```diff
@@ -232,100 +229,92 @@ where
             )
         }
 
-        loop {
```
Comment (dvdplm) [inline protocols/ping/src/dial_handler.rs]: Curious, what was the error reported here? Was the loop never looped, i.e. the code always returned?

File: stores/peerstore/src/peerstore.rs (hunk 1 above)
Comment (twittner) [reply stores/peerstore/src/peerstore.rs]: https://rust-lang.github.io/rust-clippy/list/index.html#inline_fn_without_body

File: protocols/ping/src/dial_handler.rs (hunk 2 above)
Comment (twittner) [reply protocols/ping/src/dial_handler.rs]: > Was the loop never looped [...]

Yes.

**Pull request merged**